File: server/app/models/historical_summary.py
from sqlalchemy import Column, Integer, Text, DateTime
from sqlalchemy.sql import func
from app.database import Base


class HistoricalSummary(Base):
    __tablename__ = "historical_summaries"

    id = Column(Integer, primary_key=True, index=True)
    session_id = Column(Integer, nullable=False, index=True)
    version = Column(Integer, nullable=False)
    content = Column(Text, nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)

File: server_go/internal/model/historical_summary.go
package model

import "time"

type HistoricalSummary struct {
	ID        int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	SessionID int64     `gorm:"not null;index;column:session_id" json:"session_id"`
	Version   int       `gorm:"not null" json:"version"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	Narrative string    `gorm:"type:text;not null;default:''" json:"narrative"`
	CreatedAt time.Time `gorm:"not null;autoCreateTime" json:"created_at"`
}

func (HistoricalSummary) TableName() string { return "historical_summaries" }
